project-koku/koku
# Cost 1486 ocp on all partitioned table 3

## Jira Ticket

[COST-1486](https://issues.redhat.com/browse/COST-1486)

## Dependencies

PR #3059 (Additive migration to create partitioned tables)
PR #3064 (Required for testing against a new DB)

## Description

This PR will make the code changes necessary to switch from using the materialized views:
* reporting_ocpall_compute_summary
* reporting_ocpall_cost_summary
* reporting_ocpall_cost_summary_by_account
* reporting_ocpall_cost_summary_by_region
* reporting_ocpall_cost_summary_by_service
* reporting_ocpall_database_summary
* reporting_ocpall_network_summary
* reporting_ocpall_storage_summary
* reporting_ocpallcostlineitem_daily_summary
* reporting_ocpallcostlineitem_project_daily_summary

to using the partitioned tables:
* reporting_ocpallcostlineitem_daily_summary_p
* reporting_ocpallcostlineitem_project_daily_summary_p

and the materialized views built from them:
* reporting_ocpall_compute_summary_p
* reporting_ocpall_cost_summary_p
* reporting_ocpall_cost_summary_by_account_p
* reporting_ocpall_cost_summary_by_region_p
* reporting_ocpall_cost_summary_by_service_p
* reporting_ocpall_database_summary_p
* reporting_ocpall_network_summary_p
* reporting_ocpall_storage_summary_p

## Testing
1. Start from clean DB
2. `make docker-reinitdb-with-sources`
3. `make load-test-customer-data start=<2 months ago> end=<today>
4. launch psql
5. `set search path = acct10001, public;`
6. `\dt reporting_ocpallcostlineitem_daily_summary_p reporting_ocpallcostlineitem_project_daily_summary_p` should show these tables as `partitioned table`
7. `\d+ reporting_ocpallcostlineitem_daily_summary_p reporting_ocpallcostlineitem_project_daily_summary_p` should list the table definitions as well as the partition names for each table
8. Execute a `select count(*) from` each of these tables should show data.
9. Execute a `select count(*) from` the `_p` materialized views should show that there is data present

[Smoke Tests](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/1381/console)

```
18:14:20 =========================== short test summary info ============================
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-last-30-days]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-last-month-daily]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-two-months-ago-daily]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-last-month]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-two-months-ago]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-date-min-today]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-date-45ago-today]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-last-30-days]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-last-month-daily]
18:14:20 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-two-months-ago-daily]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-last-month]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-two-months-ago]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-date-min-today]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-project-date-45ago-today]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-last-30-days]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-last-month-daily]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-two-months-ago-daily]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-two-months-ago]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-date-min-today]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-project-date-45ago-today]
18:14:21 FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_validate_total[usage-node-cluster-two-months-ago-daily]
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_tiered_rates_crud
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_negative_unbound_tier[tiered_rate_unbound_start]
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_negative_unbound_tier[tiered_rate_unbound_start_and_end]
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_negative_tiered_rate_overlap
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_negative_10_decimal_places
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_cost_type_negative
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_source_negative
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_source_update_negative
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__cost_model.py::test_api_cost_model_tag_rate[node_cost]
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_cost_reports.py::test_api_ocp_cost_group_by_exclusive_matching[project]
18:14:21 ERROR lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_azure_source_returns_cost_model
18:14:21 = 21 failed, 5782 passed, 196 skipped, 619 deselected, 26 warnings, 11 errors in 3243.87s (0:54:03) =
```
